examples/cmake_with_target/BUILD.bazel

```diff
@@ -0,0 +1,34 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+filegroup(
+    name = "srcs",
+    srcs = glob(["src/**"]),
+    visibility = ["//visibility:public"],
+)
+
+# Building all targets with the `srcs` CMake project will fail.
+# this example demonstrates how users can avoid unnecessary targets
+cmake(
+    name = "cmake_with_target",
+    build_args = [
+        # Pass arguments to the underlying build tool
+        "--",
+        "-j 1",
```

Comment:
Are we better demonstrating `--parallel` here: https://stackoverflow.com/a/50883540/6857772 or https://cmake.org/cmake/help/latest/envvar/CMAKE_BUILD_PARALLEL_LEVEL.html? Feels like a more portable way of doing this.

Reply:
My goal here wasn't strictly to demonstrate parallel builds. I just wanted to show how to pass arguments to the build tool
https://cmake.org/cmake/help/latest/command/build_command.html
note the `-- -i`

Reply:
I've updated the example to hopefully make this clearer